src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -411,7 +409,7 @@ ov::hetero::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model
         auto device_config = metaDevices[m_compiled_submodels[id].device];
         device_config[ov::cache_dir.name()] = "";
 
-        m_compiled_submodels[id].compiled_model = plugin->get_core()->compile_model(m_compiled_submodels[id].model,
+        m_compiled_submodels[id].compiled_model = plugin->get_core()->compile_model(subFunctions[id]->clone(),
```

Comment:
Why do you need to clone the model here? I think we will have another copy inside the plugin compile model

Reply:
Done in https://github.com/openvinotoolkit/openvino/pull/18576